File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    def compare(a, b):
        if a + b > b + a:
            return -1
        elif a + b < b + a:
            return 1
        else:
            return 0
    
    str_numbers = list(map(str, numbers))
    
    str_numbers.sort(key=cmp_to_key(compare))
    
    answer = ''.join(str_numbers)
    
    return str(int(answer))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        PriorityQueue<String> pq = new PriorityQueue<>((a, b) ->{
            return (b + a).compareTo(a + b);
        });
        
        for(int num : numbers){
            pq.add(String.valueOf(num));
        }
        
        StringBuilder sb = new StringBuilder();
        
        while(!pq.isEmpty()){
            sb.append(pq.poll());
        }
        
        String answer = sb.toString();
        return answer.charAt(0) == '0' ? "0" : answer;
    }
}